src/Core/Resolvers/SqlPaginationUtil.cs

```diff
@@ -234,45 +260,56 @@ public static JsonDocument CreatePaginationConnectionFromJsonDocument(JsonDocume
             try
             {
                 afterJsonString = Base64Decode(afterJsonString);
-                after = JsonSerializer.Deserialize<IEnumerable<PaginationColumn>>(afterJsonString);
+                requestAfter = JsonSerializer.Deserialize<IEnumerable<NextLinkColumn>>(afterJsonString);
 
-                if (after is null)
+                if (requestAfter is null)
                 {
                     throw new ArgumentException("Failed to parse the pagination information from the provided token");
                 }
 
                 Dictionary<string, PaginationColumn> afterDict = new();
-                foreach (PaginationColumn column in after)
+                foreach (NextLinkColumn column in requestAfter)
                 {
                     // REST calls this function with a non null sqlMetadataProvider
                     // which will get the exposed name for safe messaging in the response.
                     // Since we are looking for pagination columns from the $after query
                     // param, we expect this column to exist as the $after query param
                     // was formed from a previous response with a nextLink. If the nextLink
                     // has been modified and backingColumn is null we throw exception.
-                    string backingColumnName = GetBackingColumnName(entityName, column.ColumnName, sqlMetadataProvider);
+                    string backingColumnName = GetBackingColumnName(entityName, column.FieldName, sqlMetadataProvider);
                     if (backingColumnName is null)
                     {
-                        throw new DataApiBuilderException(message: $"Cursor for Pagination Predicates is not well formed, {column.ColumnName} is not valid.",
-                                                       statusCode: HttpStatusCode.BadRequest,
-                                                       subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+                        throw new DataApiBuilderException(
+                            message: $"Cursor for Pagination Predicates is not well formed, {column.FieldName} is not valid.",
+                            statusCode: HttpStatusCode.BadRequest,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
                     }
 
+                    PaginationColumn pageColumn = new(
+                        tableName: "",
+                        tableSchema: "",
+                        columnName: backingColumnName,
+                        value: column.Value,
+                        paramName: column.ParamName,
+                        direction: column.Direction);
+                    after.Add(pageColumn);
                     // holds exposed name mapped to exposed pagination column
-                    afterDict.Add(column.ColumnName, column);
+                    afterDict.Add(column.FieldName, pageColumn);
```

Comment:
can you give a more descriptive name to `afterDict` ? i get that the name already existed before this pr, but along with your other collections with the `after` name, then these all get confusing about what their purpose is.

Reply:
cleaned up the names